org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/id/IdCodecTest.java

```diff
@@ -10,520 +10,15 @@
-    }
-  }
-
-  @IdTypeName("scout.FixtureCompositeWithNullValuesId")
-  protected static final class FixtureCompositeWithNullValuesId extends AbstractCompositeId {
-    private static final long serialVersionUID = 1L;
-
-    private FixtureCompositeWithNullValuesId(FixtureStringId c1, FixtureUuId c2) {
-      super(c1, c2);
-    }
-
-    @RawTypes
-    public static FixtureCompositeWithNullValuesId of(String c1, UUID c2) {
-      if (StringUtility.isNullOrEmpty(c1) && c2 == null) {
-        return null;
-      }
-      return new FixtureCompositeWithNullValuesId(FixtureStringId.of(c1), FixtureUuId.of(c2));
-    }
-
-    public static FixtureCompositeWithNullValuesId of(FixtureStringId c1, FixtureUuId c2) {
-      if (c1 == null && c2 == null) {
-        return null;
-      }
-      return new FixtureCompositeWithNullValuesId(c1, c2);
-    }
-  }
-
-  @IdTypeName("scout.FixtureLocaleId")
-  protected static final class FixtureLocaleId extends AbstractRootId<Locale> {
-    private static final long serialVersionUID = 1L;
-
-    private FixtureLocaleId(Locale id) {
-      super(id);
-    }
-
-    public static FixtureLocaleId of(Locale locale) {
-      if (locale == null) {
-        return null;
-      }
-      return new FixtureLocaleId(locale);
-    }
-  }
-
-  protected static final class CustomComparableRawDataType implements Comparable<CustomComparableRawDataType> {
-    private final int m_value;
-
-    private CustomComparableRawDataType(int value) {
-      m_value = value;
-    }
-
-    static CustomComparableRawDataType of(String value) {
-      return new CustomComparableRawDataType(Integer.parseInt(value));
-    }
-
-    @Override
-    public String toString() {
-      return "" + m_value;
-    }
-
-    @Override
-    public int compareTo(CustomComparableRawDataType o) {
-      return m_value - o.m_value;
-    }
-
-    @Override
-    public boolean equals(Object o) {
-      if (this == o) {
-        return true;
-      }
-      if (o == null || getClass() != o.getClass()) {
-        return false;
-      }
-      CustomComparableRawDataType that = (CustomComparableRawDataType) o;
-      return m_value == that.m_value;
-    }
-
-    @Override
-    public int hashCode() {
-      return Objects.hash(m_value);
-    }
-  }
-
-  @IdTypeName("scout.FixtureCustomComparableRawDataId")
-  protected static final class FixtureCustomComparableRawDataId extends AbstractRootId<CustomComparableRawDataType> {
-    private static final long serialVersionUID = 1L;
-
-    private FixtureCustomComparableRawDataId(CustomComparableRawDataType id) {
-      super(id);
-    }
+public class IdCodecTest extends AbstractIdCodecTest {
 
-    public static FixtureCustomComparableRawDataId of(CustomComparableRawDataType date) {
-      if (date == null) {
-        return null;
-      }
-      return new FixtureCustomComparableRawDataId(date);
-    }
+  @Override
+  protected IdCodec getCodec() {
+    return BEANS.get(IdCodec.class);
```

Comment:
Always get the instance again or use member variable instead?

Reply:
Done